fatrace.c

```diff
@@ -234,6 +315,8 @@ static void
 print_event (const struct fanotify_event_metadata *data,
              const struct timeval *event_time)
 {
+    const char* problems[7];
```

Comment:
statically sized array without a constant and bounds checks. *if* you need this, please add an `add_problem()` function which keeps a counter and uses `realloc()`. But putting error messages into json is weird -- these just belong to stderr IMHO.

Also, these are currently debug messages in CLI mode, so "not expected" or "not interesting". If you encounter them in real life, can you please file an issue about it and then we can see case-by-case how to handle them more verbosely also with the CLI? Lumping them at the end of the jsonl isn't useful anyway -- it's hard to correlate them to the event that triggered them.

Reply:
max size 1000 and erroring if it is reached is fine for me, thanks. I still disagree with collecting them into the JSON in the first place, but I have a much less strong opinion about that.

> Including the problems in the JSON object makes it trivial to correlate them to the event that triggered them, 

Sorry, I misunderstood the core before. `problems` is not a global top-level entry but each event has its own. That's fine.

Reply:
As this is pushed to #51, does that resolve this conversation?